### PR TITLE
SNOW-1728340: support gcp and azure

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

4. (Optional) PR for stored-proc connector:

### CODE REVIEW

File: src/snowflake/connector/aio/_azure_storage_client.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        )
+        end_point: str = stage_info["endPoint"]
+        if end_point.startswith("blob."):
+            end_point = end_point[len("blob.") :]
+        self.endpoint = end_point
+        self.storage_account: str = stage_info["storageAccount"]
+        self.azure_location = self.extract_container_name_and_path(
+            stage_info["location"]
+        )
+        self.block_ids: list[str] = []
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/aio/_azure_storage_client.py:64]: we can just call `SnowflakeAzureRestClientSync.__init__`

File: src/snowflake/connector/aio/_s3_storage_client.py [hunk 2]
```diff
@@ -34,7 +34,8 @@
 from ._storage_client import SnowflakeStorageClient as SnowflakeStorageClientAsync
 
 if TYPE_CHECKING:  # pragma: no cover
-    from ._file_transfer_agent import SnowflakeFileMeta, StorageCredential
+    from ..file_transfer_agent import StorageCredential
+    from ._file_transfer_agent import SnowflakeFileMeta
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/aio/_s3_storage_client.py:38]: let's import both from sync

**Comment by sfc-gh-aling** [comment]
it seems like all asyncio gcp tests got stuck

**Comment by sfc-gh-yuwang** [comment]
> Details

Yes, I noticed that GCP has taken very long time to finish

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**